### PR TITLE
Managed Files API - Second Pass

### PR DESCRIPTION
**Pull request by danderson00**
This PR 
- addresses previous PR feedback
- introduces default implementations and relevant extension methods to platform specific libraries
- allows consumers to pass in a base path for file storage location
- introduces MobileServiceFileSystemFile that exposes the physical path for files

### CODE REVIEW

**Comment by danderson00** [comment]
@fabiocav

File: src/Microsoft.WindowsAzure.Mobile.Files/Managed/LocalStorage/ILocalStorageProvider.cs [hunk 1]
```diff
@@ -12,5 +12,7 @@ public interface ILocalStorageProvider
         Task AddAsync(MobileServiceFile file, Stream source);
         Task<Stream> GetAsync(MobileServiceFile file);
         Task DeleteAsync(MobileServiceFile file);
+
```
Comment (fabiocav) [inline src/Microsoft.WindowsAzure.Mobile.Files/Managed/LocalStorage/ILocalStorageProvider.cs]: nit: extra spaces

Comment (danderson00) [inline src/Microsoft.WindowsAzure.Mobile.Files/Managed/LocalStorage/ILocalStorageProvider.cs]: For some reason, that is an outdated diff. It's already fixed, I don't know why that is showing.

Comment (fabiocav) [inline src/Microsoft.WindowsAzure.Mobile.Files/Managed/LocalStorage/ILocalStorageProvider.cs]: Ah, because I'm looking at this specific commit. 👍

**Pull request merged**